### DISCUSSION TITLE
Variables don't work in quoted topic names

### DISCUSSION THREAD

**Discussion by jameskuszmaul-brt** [foxglove/community#306] [Q&A]
**Description**

I've got topic names that require escaping (which was solved by foxglove/studio#2891), but I don't seem to be able to use variables in my topic names (I don't know if variables are supposed to be usable in normal topic names either, admittedly). I'd like to be able to do something like:

`"/\$machine_name/topic name".xyz` (probably would actually want something like `"/${machine_name}/topic name".xyz` for use with variables, but then have `"/\$\{machine_name\}/topic name".xyz` be what you use when you have literal `${}` characters in your topic names).

**Steps To Reproduce**

Can provide a bag file as needed--although the MCAP files referenced in https://github.com/foxglove/studio/issues/2886#issuecomment-1047432246 should be good for testing this.

**Expected Behavior**

**Actual Behavior**

**Comment by jtbandes** [comment]
Converting to a feature request – variables don't work in topic names currently, only filters.

**Comment by jtbandes** [comment]
Related: 
- https://github.com/foxglove/community/issues/185

**Comment by jameskuszmaul-brt** [comment]
The glob pattern idea referenced in foxglove/community#185 would also resolve the core use-case referenced here, although I could still generate some corner-cases where variable names in topics would be helpful.
